app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,90 @@
+class RecipesController < ApplicationController
+  # GET /recipes
+  def index
+    @recipes = Recipe.all
+  end
+
+  # GET /recipes/1
+  def show
+    @recipe = Recipe.find(params[:id])
+  end
+
+  # GET /recipes/new
+  def new
+    @recipe = Recipe.new
+  end
+
+  def create
+    @recipe = current_user.recipes.new(recipe_params)
+    respond_to do |f|
+      if @recipe.save
+        f.html { redirect_to my_recipes_path, notice: 'Recipe was successfully created' }
+      else
+        f.html { render :new }
+      end
+    end
+  end
+
+  def destroy
+    @recipe = Recipe.find(params[:id]).destroy
+    respond_to do |f|
+      f.html { redirect_to my_recipes_path, notice: 'Recipe was successfully destroyed' }
+    end
+  end
+
+  def my_recipes
+    @recipes = current_user.recipes.includes(:user)
+  end
+
+  def public_recipes
+    @recipes = Recipe.includes(:user).where(public: true)
+
+    @shopping_lists = {}
+    @recipes.each do |recipe|
+      @shopping_lists[recipe.id] = generate_shopping_list(recipe)
+    end
```

Comment:
- [ ] Upon creating a public recipe successfully, clicking on the link to go to the public recpies list produces the following error
![image](https://github.com/chiztechnology/rails-capstone-project/assets/81932478/f4e9fd12-7338-4f57-85ab-90e98cf047a8)
- [ ] As requested by the previous reviewer kindly fix all your technical functionality to work as expected